File: client/src/types/JsonModels/WorkoutTemplates/WorkoutTemplateExerciseGroupModel.ts
import type { JsonModels } from "../../backend";
import type { ExerciseGroupType } from "../Enums/ExerciseGroupType";

export interface WorkoutTemplateExerciseGroupModel
{
	id: number;
	sortOrder: number;
	groupType: ExerciseGroupType;
	restBetweenExercisesSeconds?: number;
	restAfterGroupSeconds?: number;
	rounds: number;
	exercises: JsonModels.WorkoutTemplates.WorkoutTemplateExerciseModel[];
}

File: server/FitMate.Core/JsonModels/WorkoutTemplates/WorkoutTemplateExerciseGroupModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.WorkoutTemplates;

public class WorkoutTemplateExerciseGroupModel
{
    public long Id { get; set; }
    public int SortOrder { get; set; }
    public ExerciseGroupType GroupType { get; set; } = ExerciseGroupType.Straight;
    public int? RestBetweenExercisesSeconds { get; set; }
    public int? RestAfterGroupSeconds { get; set; }
    public int Rounds { get; set; }
    public List<WorkoutTemplateExerciseModel> Exercises { get; set; } = [];
}
